rocketpy/simulation/monte_carlo.py

```diff
@@ -159,22 +197,360 @@ def simulate(self, number_of_simulations, append=False):
+        ----------
+            append: bool
+                If True, the results will be appended to the existing files. If
+                False, the files will be overwritten.
+            light_mode: bool
+                If True, only variables from the export_list will be saved to
+                the output file as a .txt file. If False, all variables will be
+                saved to the output file as a .h5 file.
+
+        Returns
+        -------
+        None
+        """
+        # Create data files for inputs, outputs and error logging
+        open_mode = "a" if append else "w"
+
+        # Open files
+        if light_mode:
+            # get initial simulation index
+            idx_i, idx_o = self.__get_light_indexes(
+                self._input_file, self._output_file, append=append
+            )
+
+            # open files in write/append mode
+            input_file = open(self._input_file, open_mode, encoding="utf-8")
+            output_file = open(self._output_file, open_mode, encoding="utf-8")
+            error_file = open(self._error_file, open_mode, encoding="utf-8")
+        else:
+            input_file = h5py.File(Path(self._input_file).with_suffix(".h5"), open_mode)
+            output_file = h5py.File(
+                Path(self._output_file).with_suffix(".h5"), open_mode
+            )
+            error_file = open(self._error_file, open_mode, encoding="utf-8")
+
+            idx_i = self.__get_initial_sim_idx(input_file, light_mode=light_mode)
+            idx_o = self.__get_initial_sim_idx(output_file, light_mode=light_mode)
+
+        if idx_i != idx_o:
+            raise ValueError(
+                "Input and output files are not synchronized. Append mode is not available."
+            )
+
+        # Run simulations
         try:
             while self.iteration_count < self.number_of_simulations:
-                self.__run_single_simulation(input_file, output_file)
+                self.__run_single_simulation(
+                    self.iteration_count + idx_i,
+                    input_file,
+                    output_file,
+                    light_mode=light_mode,
+                )
+
         except KeyboardInterrupt:
             print("Keyboard Interrupt, files saved.")
             error_file.write(json.dumps(self._inputs_dict, cls=RocketPyEncoder) + "\n")
             self.__close_files(input_file, output_file, error_file)
+
         except Exception as error:
             print(f"Error on iteration {self.iteration_count}: {error}")
             error_file.write(json.dumps(self._inputs_dict, cls=RocketPyEncoder) + "\n")
             self.__close_files(input_file, output_file, error_file)
             raise error
 
-        self.__finalize_simulation(input_file, output_file, error_file)
+        self.__finalize_simulation(
+            input_file, output_file, error_file, light_mode=light_mode
+        )
+
+    def __run_in_parallel(self, append, light_mode, n_workers=None):
```

Comment:
What happens is the user stops the execution (Keyboard interrupt) before all the iterations be completed. Do we loose all the simulations results? 

Needs investigation.

Reply:
The workers stop exactly where they are. All complete simulations are preserved, but it is possible that a simulation only saves the input and not the output.